src/viking.js

```diff
@@ -1,14 +1,115 @@
 // Soldier
-class Soldier {}
+
+class Soldier {
+  constructor (health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack () {
+    return this.strength;
+  }
+
+  receiveDamage (theDamage){
+    this.health -= theDamage;
+    }
+}
+
+
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+   constructor (name, health, strength){
+    super (health, strength);
+    this.name = name;
+   }
+   receiveDamage (theDamage){
+    this.health -= theDamage;
+    if (this.health > 0){
+      return `${this.name} has received ${theDamage} points of damage`;
+    }
+    else {
+      return `${this.name} has died in act of combat`;
+    }
+    }
+
+    battleCry () {
+      return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+
+class Saxon extends Soldier {
+   constructor (health,strength){
+    super(health, strength);
+   } 
+
+   receiveDamage (theDamage){
+    this.health -= theDamage;
+    if (this.health > 0){
+      return `A Saxon has received ${theDamage} points of damage`;
+    }
+    else {
+      return `A Saxon has died in combat`;
+    }
+    }
+}
 
 // War
-class War {}
+class War {
+  constructor(){
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(vicking) {
+    this.vikingArmy.push(vicking);
+  }
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+  
+  vikingAttack() {
+    
+   
+    let resultAttack = (this.saxonArmy[Math.floor(Math.random()*this.saxonArmy.length)]).receiveDamage((this.vikingArmy[Math.floor(Math.random()*this.vikingArmy.length)]).attack());
+    /* for (i=0; i<this.saxonArmy.length; i++){
+      if (this.saxonArmy[i].health <= 0) {
+        this.saxonArmy.splice(this.saxonArmy[i],1);
+        return this.saxonArmy;
+        } */
```

Comment:
Here you had the perfect logic, the only thing is the "return this.saxonArmy", inside of the loop you shouldn't return anything meaning that your logic was working but you were return the army itself